src/js/components/Tabs/Tabs.js

```diff
@@ -39,6 +57,74 @@ const Tabs = forwardRef(
     delete rest.onActive;
     /* eslint-enable no-param-reassign */
 
+    const tabRefs = React.Children.map(children, () => React.createRef());
+
+    // check if tab is in view
+    function isVisible(element) {
+      const tabRect = element?.getBoundingClientRect();
+      const headerRect = targetRef.current?.getBoundingClientRect();
+      return (
+        tabRect.left >= headerRect.left - 1 &&
+        tabRect.right <= headerRect.right + 1
+      );
+    }
+
+    const updateArrowState = useCallback(() => {
+      setDisableLeftArrow(isVisible(tabRefs[0].current));
+      setDisableRightArrow(isVisible(tabRefs[tabRefs.length - 1].current));
+    }, [tabRefs]);
+
+    useLayoutEffect(() => {
+      if (scroll && overflow) {
+        if (!isVisible(tabRefs[activeIndex].current)) {
+          // if the active tab isn't visible scroll to it
+          tabRefs[activeIndex].current.scrollIntoView();
+          updateArrowState();
+        }
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [activeIndex, overflow, scroll]);
+
+    useEffect(() => {
+      // if the focus index is changing make sure the previous and next arrows
+      // are in sync
+      if (scroll && overflow && focusIndex) updateArrowState();
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [focusIndex, overflow, scroll]);
+
+    useLayoutEffect(() => {
+      if (scroll) {
+        const onResize = () => {
+          // check if tabs are overflowing
+          if (targetRef.current.scrollWidth > targetRef.current.offsetWidth) {
+            setOverflow(true);
+          } else setOverflow(false);
+          updateArrowState();
+        };
+
+        if (
+          overflow === undefined ||
+          disableLeftArrow === undefined ||
+          disableRightArrow === undefined
+        ) {
+          onResize();
+        }
```

Comment:
When I refresh the page, the responsive controls aren't visible until I resize the page. Do we need to always call `onResize` on the initial render (outside of this condition)

https://user-images.githubusercontent.com/12522275/169605873-3f400d21-2bfc-4e49-9380-0094cf8f3cb8.mov

?

Reply:
Hmm I haven't been able to reproduce this behavior. Is it happening every time you are refreshing the page or is it only happening sometimes?